### DISCUSSION TITLE
How to migrate from V4 to V5?

### DISCUSSION THREAD

**Discussion by baybal** [adonisjs/core#3274] [Help]
Adonis 5 feels almost like a brand new framework, with tons of code having to be rewritten.

Any way to do a staged transition, or at least some methodologic guide how to survive move from 4 to 5 while in production?

V4 support period ends this December, but the state of docs, and everything wasn't that good on 4.1 to begin with. I don't think I can stay on V4.1 long term, as I am doing more fixes to it than making my own code by now.

**Comment by RomainLanz** [comment]
Hey @baybal! 👋🏻 

The transition should be simple, it all depends on the size of your project.

I would recommend to start a brand new project with the V5 scaffolding tools and start to copy/paste your logic there.
Few things change between version 4 and version 5, the big change is that you will be using TypeScript instead of JavaScript.

Do not hesitate to ask more questions during this transition.

**Comment by D40298008** [comment]
I am trying to migrate from 4.1 to 5 here in 2023.  I am trying to figure out how the migrations would work with the old V4 (or a clone of the old V4 database).

I have created a completely new project and I was thinking of copying all the migration files (updating to TS obviously) with the exact timestamps and everything.

However, the old V4 migrations have a name column like `1503250034168_user` and the new V5 ones have a name column like `database/migrations/1503250034168_user`.

I could probably update them manually, but I was just wondering if there was a basic guide anywhere on the 4 to 5 migration process.  I can't seem to find one anywhere.

**Reply by EmilyRagan** [reply]
I am having the same issue

**Reply by RomainLanz** [reply]
You should change them by hand 👍🏻

**Reply by EmilyRagan** [reply]
@RomainLanz I was really hoping for a way to configure V5 to recognize V4 migration names, or perhaps instructions for an additional step that could be taken to rename the existing migrations that were run in V4 before attempting to run them from V5. I am doing an upgrade across many machines that I do not have access to, so they cannot be changed by hand.

**Reply by RomainLanz** [reply]
We have a versioning system for migrations in place, but I don't believe the change from V4 to V5 was considered or if the system was already in place when V4 existed. Maybe @thetutlage has more information on this.
